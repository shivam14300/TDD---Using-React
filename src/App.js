import React from 'react';
import fetch from 'node-fetch';
const {fetchData} = require ('./http');

export default class App extends React.Component {
  constructor () {
    super ();

    this.state = {
      title1: 'API1',
      title2: 'API2',
      title3: 'API3',
      title4: 'API4',
      data: {movies: [{id: 'not set'}]},
      data1: {id: ['not set']},
      data2: {movies: [{releaseYear: 'not set'}]},
    };

    this.changeTitle1 = this.changeTitle1.bind (this);
    this.changeTitle2 = this.changeTitle2.bind (this);
    this.changeTitle3 = this.changeTitle3.bind (this);
    this.changeTitle4 = this.changeTitle4.bind (this);
  }

  async API_POST () {
    return fetch ('https://jsonplaceholder.typicode.com/posts', {
      method: 'POST',
      body: JSON.stringify ({
        title: 'foo',
        body: 'bar',
        userId: 1,
      }),
      headers: {
        'Content-type': 'application/json; charset=UTF-8',
      },
    })
      .then (response => response.json ())
      .then (res => {
        this.setState ({data1: res});
        return res;
      });
  }

  async API_GET () {
    return fetch ('http://reactnative.dev/movies.json')
      .then (result => {
        return result.json ();
      })
      .then (res => {
        this.setState ({data: res});
        return res;
      });
  }

  async API_GET1 () {
    return fetch ('http://reactnative.dev/movies.json')
      .then (result => {
        return result.json ();
      })
      .then (res => {
        this.setState ({data2: res});
        return res;
      });
  }

  async loadTitle () {
    return fetchData ().then (ext => {
      const title = ext.title;
      const upper_title = title.toUpperCase ();
      return upper_title;
    });
  }

  changeTitle1 () {
    this.setState ({title1: 'Calling API1...'});
    // this.setState({ data: {"movies": [{"id" : "1"},]} });
    return this.API_GET ();
  }

  changeTitle2 () {
    this.setState ({title2: 'Calling API2...(Please Wait)'});
    this.API_POST ();
  }

  changeTitle3 () {
    this.setState ({title3: 'Calling API3...'});
    this.API_GET1 ();
  }

  changeTitle4 () {
    this.loadTitle ().then (title => {
      console.log (title);
    });
  }

  render () {
    return (
      <div>
        <button id="button1" onClick={this.changeTitle1}>
          {this.state.title1}
        </button>
        <br />
        <p id="para">
          {this.state.data.movies[0].id}
        </p>
        <button id="button2" onClick={this.changeTitle2}>
          {this.state.title2}
        </button>
        <br />
        <p>
          {this.state.data1.id}
        </p>
        <button id="button3" onClick={this.changeTitle3}>
          {this.state.title3}
        </button>
        <br />
        <p style={{color: 'red'}}>
          {this.state.data2.movies[0].releaseYear}
        </p>
        <button id="button4" onClick={this.changeTitle4}>
          {this.state.title4}
        </button>
        <br />
      </div>
    );
  }
}
