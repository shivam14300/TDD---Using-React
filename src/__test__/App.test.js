// jest.mock('../http.js');

import React from 'react';
import App from '../App';
import { shallow } from 'enzyme';
import '../setupTests';
import renderer from 'react-test-renderer';

test ('test snapshot', () => {
  const component = renderer.create(<App/>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
}) 


// describe('test', () => {
//   it('test1', () => {
//     const res = new App();
//     res.loadTitle().then(title => {
//       expect(title).toEqual('DELECTUS AUT AUTEM');
//     })
//   })
// })

// describe('Testing Button Clicks',() =>{
//   let wrapper;
//   it('Testing Button Click of API1',() => {
//     wrapper = shallow(<App/>);
//     wrapper.find('#button1').simulate('click');
//     wrapper.update();
//     expect(wrapper.state('title1')).toEqual('Calling API1...');
//   })
//   it('Testing Button Click of API2',() => { 
//     wrapper = shallow(<App/>);
//     wrapper.find('#button2').simulate('click');
//     wrapper.update();
//     expect(wrapper.state('title2')).toEqual('Calling API2...(Please Wait)');
//   })
//   it('Testing Button Click of API3',() => {
//     wrapper = shallow(<App/>);
//     wrapper.find('#button3').simulate('click');
//     wrapper.update();
//     expect(wrapper.state('title3')).toEqual('Calling API3...');
//   })
// })


// describe('Testing API function',() => {
  
//   const fake = jest.spyOn(App.prototype, 'API_GET');
//   const wrapper = shallow(<App/>);
//   wrapper.find('#button1').simulate('click');
//   wrapper.update();
//   it('check that function API_GET is defined', () => {
//     expect(fake).toBeDefined(); // check that function API_GET is defined
//   })
  
//   it('check that function API_GET is called on click', () => {
//     expect(fake).toHaveBeenCalled(); // check that function API_GET is called
//   })
  
//   it('check that function API_GET returns successfully on calling', () => {
//     expect(fake).toHaveReturned(); // check that function API_GET returns successfully
//   })
  
// })

// describe('Testing methods of API call using Async functions', () => {
//   it('Api fetch testing for GET',async () => {
//     const res = new App();
//     var data = await res.API_GET();
//     expect(data.movies[0].id).toEqual('1');   
//     expect(data.movies[1]).toHaveProperty(['releaseYear'],'1985');   
//   },100000)
  
//   it('Api fetch testing for POST',async () => {
//     const res = new App();
//     var data = await res.API_POST();
//     expect(data.id).toEqual(101);   
//   },100000000)
  
  
// })

// describe('Test rendering', () => {
//   it('Test that text is rendered', () => {
//     const wrapper = shallow(<App/>);
//     expect(wrapper.find('#para').text()).toEqual("not set")
//     wrapper.find('#button1').simulate('click');
//     wrapper.update();
//     // expect(wrapper.find('#para').text()).toEqual("1")
//   })
// })