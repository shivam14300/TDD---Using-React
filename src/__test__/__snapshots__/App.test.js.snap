// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div>
  <button
    id="button1"
    onClick={[Function]}
  >
    API1
  </button>
  <br />
  <p
    id="para"
  >
    not set
  </p>
  <button
    id="button2"
    onClick={[Function]}
  >
    API2
  </button>
  <br />
  <p>
    not set
  </p>
  <button
    id="button3"
    onClick={[Function]}
  >
    API3
  </button>
  <br />
  <p>
    not set
  </p>
  <button
    id="button4"
    onClick={[Function]}
  >
    API4
  </button>
  <br />
</div>
`;
